peerstore/peerstore.go

```diff
@@ -233,5 +233,12 @@ type ProtoBook interface {
 	AddProtocols(peer.ID, ...string) error
 	SetProtocols(peer.ID, ...string) error
 	RemoveProtocols(peer.ID, ...string) error
+
+	// SupportsProtocols returns the set of protocols the peer supports from among the given protocols.
+	// If the returned error is not nil, the result is indeterminate.
 	SupportsProtocols(peer.ID, ...string) ([]string, error)
+
+	// SupportsAnyProtocol returns true if the peer supports ATLEAST ONE of the given protocols and false otherwise.
+	// If the returned error is not nil, the result is indeterminate.
+	SupportsAnyProtocol(peer.ID, ...string) (bool, error)
```

Comment:
What about `FirstSupportedProtocol(peer.ID, ...string) (string, error)`? This is what you want 99.9% of the time anyways and covers the new stream use-case.

Reply:
@Stebalien Great idea !